java/core/src/java/org/apache/orc/impl/DynamicByteArray.java

```diff
@@ -59,10 +60,16 @@ private void grow(int chunkIndex) {
         int newSize = Math.max(chunkIndex + 1, 2 * data.length);
         data = Arrays.copyOf(data, newSize);
       }
-      for(int i=initializedChunks; i <= chunkIndex; ++i) {
+      for (int i = initializedChunks; i <= chunkIndex; ++i) {
         data[i] = new byte[chunkSize];
       }
       initializedChunks = chunkIndex + 1;
+    } else if (chunkIndex < 0) {
```

Comment:
How many places do we have like the following?
```
grow((length + valueLength) / chunkSize);
```

Reply:
If NPE happens always before, this looks like a precondition check for this method.
- Shall we place this at the beginning of this method as an independent `if` statement?
- Could you add your new logic to the function description too? The current one looks vague.
https://github.com/apache/orc/blob/d403e11b7eab126259fb0683c0eb5c6131fd347b/java/core/src/java/org/apache/orc/impl/DynamicByteArray.java#L54
```

BTW, I revise the PR title, @cxzl25 . You can change back if you want.